lang/test/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransformTests.kt

```diff
@@ -0,0 +1,152 @@
+package org.partiql.lang.planner.transforms
+
+import com.amazon.ion.system.IonSystemBuilder
+import com.amazon.ionelement.api.ionBool
+import com.amazon.ionelement.api.ionInt
+import com.amazon.ionelement.api.ionString
+import com.amazon.ionelement.api.toIonValue
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.partiql.lang.domains.PartiqlLogical
+import org.partiql.lang.domains.id
+import org.partiql.lang.domains.pathExpr
+import org.partiql.lang.syntax.SqlParser
+import org.partiql.lang.util.ArgumentsProviderBase
+import org.partiql.lang.util.SexpAstPrettyPrinter
+
+/**
+ * Test cases in this class might seem a little light--that's because [AstToLogicalVisitorTransform] is getting
+ * heavily exercised during many other integration tests.  These should be considered "smoke tests".
+ */
+class AstToLogicalVisitorTransformTests {
+    private val ion = IonSystemBuilder.standard().build()
+    private val parser = SqlParser(ion)
+
+    private fun parseAndTransform(sql: String): PartiqlLogical.Statement {
+        val parseAstStatement = parser.parseAstStatement(sql)
+        println(SexpAstPrettyPrinter.format(parseAstStatement.toIonElement().asAnyElement().toIonValue(ion)))
+        return parseAstStatement.toLogicalPlan().stmt
+    }
+
+    data class TestCase(val sql: String, val expectedAlgebra: PartiqlLogical.Statement)
+
+    private fun runTestCase(tc: TestCase) {
+        val algebra = assertDoesNotThrow("Parsing TestCase.sql should not throw") {
+            parseAndTransform(tc.sql)
+        }
+        println(SexpAstPrettyPrinter.format(algebra.toIonElement().asAnyElement().toIonValue(ion)))
+        Assertions.assertEquals(tc.expectedAlgebra, algebra)
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(ArgumentsForToLogicalTests::class)
+    fun `to logical`(tc: TestCase) = runTestCase(tc)
+
+    class ArgumentsForToLogicalTests : ArgumentsProviderBase() {
+        override fun getParameters() = listOf(
```

Comment:
Maybe it's good to also add test cases for `LET`, `OFFSET` & `LIMIT` clause?

Reply:
From the kdoc above:

> Test cases in this class might seem a little light--that's because [AstToLogicalVisitorTransform] is getting heavily exercised during many other integration tests.  These should be considered "smoke tests".

Detailed test cases were not intended here.  We have tests which run `LET`, `OFFSET` and `LIMIT` e2e.  I lean against exhaustive testing of every unit in this type of project because the integration tests and the e2e behavior they test is what has the most value.  Also, detailed tests like this increase maintenance overhead since any changes to those nodes means that this is yet another place that needs to be updated.